### PR TITLE
code cleanup - code to interface, add @Override, remove unused import, etc

### PR DESCRIPTION
**Pull request by hansenmc**
I was poking around and made some small improvements. No changes to functionality, just making things a little nicer.

- Removed unused imports
- Changed to use interfaces for left hand side of object assigment
(i.e. List<String> foo = new ArrayList<>() )
- Removed unnecessary modifiers for interface member variables and
methods (they are always public)
- Added missing @Override annotations
- Use try-with-resources to auto-close
- Replaced StringBuffer with StringBuilder

### CODE REVIEW

**Comment by sammefford** [comment]
Mads, thanks for taking the time to do cleanup!  It's helpful for me to see some of your improvements.  I hadn't even used try-with-resources before...somehow certain new java features evade my radar.  Also, I'm baffled that I made the mistake of comparing strings with ==  good catch fixing that!

Before I merge things in, any chance you could redo https://github.com/marklogic/data-movement/pull/108/commits/ecbb36dcc2d1dfc629306698cb16fa5caf5b4716 so it doesn't change the indenting?  I can't see what you changed because of the simultaneous whitespace changes.  Or, if you think whitespace changes are in order, could you make those in a whitespace-only commit?

Also, same thing on two files from https://github.com/marklogic/data-movement/pull/108/commits/9bd5c214a9d98d97e4085768a54c3666805355e8
  * ExportToWriterListener.java
  * QueryHostException.java

**Comment by hansenmc** [comment]
Sure, I can adjust them. I tried to fight the urge to reformat, but have a habit of applying. What is your indent/formatting pattern? 

Meantime, you can ignore whitespaces in the Github diff view by appending "?w=1" to the querystring:
- https://github.com/marklogic/data-movement/pull/108/commits/ecbb36dcc2d1dfc629306698cb16fa5caf5b4716?w=1
- https://github.com/marklogic/data-movement/pull/108/commits/9bd5c214a9d98d97e4085768a54c3666805355e8?w=1

**Comment by sammefford** [comment]
I've been using two spaces on this project . . . and you can see the rest in the files.  But many of the big reformatting from your commits is in datamovement-functionaltest which is owned by QA, and we haven't tried to sync on formatting style.  So in general my rule of thumb is stay consistent with what's already in the file.  And make white-space changes when not making other changes to keep the commit history simple and easy-to-understand.  That ?w=1 trick is cool.  But when I look at it I'm still overwhelmed by the many white-space changes due to method opening curly braces moving.

**Comment by hansenmc** [comment]
Okay, the diffs should be much cleaner now. Sorry about that.

File: data-movement-functionaltests/src/test/java/com/marklogic/datamovement/functionaltests/BasicJavaClientREST.java [hunk 1]
```diff
@@ -1553,16 +1552,16 @@ public String convertXMLEventReaderToString(XMLEventReader fileRead) throws IOEx
...
-        String str = null;
+        StringBuilder str = new StringBuilder();
 		while (reader.hasNext())
 		{
 		    reader.next();
 		    int a = reader.getEventType();
 		    if (reader.hasText())
-		    	if ( reader.getText() != "null" )
-		        str =str + reader.getText().trim();
+		    	if ( reader.getText() != "null" ) //TODO verify whether this should be performing a null check or really compare literal 'null'
```
Comment (hansenmc) [inline data-movement-functionaltests/src/test/java/com/marklogic/datamovement/functionaltests/BasicJavaClientREST.java]: @sammefford be sure to take a look at this. I wasn't sure if the intent was to do a null check, or look for the literal string value "null" in the text.

**Comment by hansenmc** [comment]
Also, `com.marklogic.datamovement.functionaltests.BasicJavaClientREST.java` is virtually identical `com.marklogic.datamovement.functionaltests.util.BasicJavaClientREST.java`. Similar with `com.marklogic.datamovement.functionaltests.ConnectedRESTQA` and `com.marklogic.datamovement.functionaltests.util.ConnectedRESTQA`. The difference in ConnectedRESTQA seems to be that "localhost" is hard-coded in to the XCC URI, and there are a few additional methods in `com.marklogic.datamovement.functionaltests.ConnectedRESTQA` that are not in the util.

**Comment by srinathgit** [comment]
Mads,
We am aware of this . I will work with @georgeajit  to clean that up.
Thanks

File: data-movement-functionaltests/src/test/java/com/marklogic/datamovement/functionaltests/BasicJavaClientREST.java (hunk 1 above)
Comment (sammefford) [inline data-movement-functionaltests/src/test/java/com/marklogic/datamovement/functionaltests/BasicJavaClientREST.java]: @srinathgit or @georgeajit want to weigh in on whether this is intentional?

**Pull request merged**